great_expectations/cli/toolkit.py

```diff
@@ -343,10 +350,10 @@ def load_checkpoint(
 
 def select_datasource(
     context: DataContext, datasource_name: Optional[str] = None
-) -> BaseDatasource:
+) -> Union[BaseDatasource, LegacyDatasource, None]:
```

Comment:
@Kilo59 How come we are replacing the use of `Optional` with the longer version of what seems to be the same type declaration?  Thanks.

Reply:
Answered above